claasp/cipher.py

```diff
@@ -1149,6 +1162,32 @@ def is_spn(self):
                 return False
         return set_of_components <= spn_components
 
+    def get_model(self, technique, problem):
+        """
+        Returns a model for a given technique and problem.
+
+        INPUT:
+
+          - ``technique`` -- **string** ; sat, smt, milp or cp
+          - ``problem`` -- **string** ; xor_differential, xor_linear, cipher_model (more to be added as more model types are added to the library)
+          """
+        if problem == 'xor_differential':
+            constructor_name = f'{technique[0].capitalize()}{technique[1:]}XorDifferentialModel'
+        elif problem == "xor_linear":
+            constructor_name = f'{technique[0].capitalize()}{technique[1:]}XorLinearModel'
+        elif problem == 'cipher_model':
+            constructor_name = f'{technique[0].capitalize()}{technique[1:]}CipherModel'
+
+        if technique == 'cp' and problem == 'xor_differential':
+            module_name = f'claasp.cipher_modules.models.{technique}.{technique}_models.{technique}_{problem}_trail_search_model'
+            constructor_name = 'CpXorDifferentialTrailSearchModel'
```

Comment:
Change name of CpXorDifferentialTrailSearchModel to CpXorDifferentialModel (same for the file name).
Then modify this function.

Reply:
Done